### PR TITLE
Update adobe-connect from 11,2021.4.36 to 11,2021.6.27

### PR DESCRIPTION
**Pull request by FnControlOption**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

https://helpx.adobe.com/adobe-connect/connect-releasenotes.html
https://helpx.adobe.com/adobe-connect/release-note/adobe-connect-2021-6-application-release-notes.html
https://helpx.adobe.com/adobe-connect/connect-downloads-updates.html

### CODE REVIEW

**Comment by miccal** [comment]
Ping @samford for a review.

File: Casks/adobe-connect.rb [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
   name "Adobe Connect"
   desc "Virtual meeting client"
   homepage "https://www.adobe.com/products/adobeconnect.html"
 
   livecheck do
-    skip "No version information available"
+    url "https://www.adobe.com/go/ConnectMac11Plus"
+    regex(%r{/(\d+(?:_\d+)*)/AdobeConnect[._-](\d+(?:_\d+)*)\.dmg}i)
+    strategy :header_match do |headers, regex|
+      match = headers["location"].match(regex)
```
Comment (samford) [inline Casks/adobe-connect.rb:14]:
```suggestion
    regex(%r{/(\d+(?:[._]\d+)*)/AdobeConnect[._-](\d+(?:[._]\d+)*)\.dmg}i)
    strategy :header_match do |headers, regex|
      match = headers["location"]&.match(regex)
      next [] if match.blank?

```

**Review by samford — CHANGES_REQUESTED** [review]
A couple tweaks:

* For a version that uses underscores, we generally use `[._]` instead of `_` in the `(?:[._]\d+)` part of a version regex, unless we specifically need to only match `_`. Sometimes version formats change from using underscores (or hyphens) to dots, so we use a character class that matches a dot or the current delimiter (`_`). This allows the regex to adapt to this change instead of breaking and needing to be manually updated.
* This returns an empty array if a `location` header isn't present in the response or the regex doesn't match the location URL. This is a little more graceful than allowing `#match` or `#tr` to error on a `nil` value.

**Comment by FnControlOption** [comment]
The `:header_match` strategy doesn't seem to like arrays:

> Error: adobe-connect: Version value must be a string; got a Array ([])

**Comment by samford** [comment]
> The `:header_match` strategy doesn't seem to like arrays:
> 
> > Error: adobe-connect: Version value must be a string; got a Array ([])

Ah yeah, I forgot that `HeaderMatch` only accepts a string from a `strategy` block and can handle a `nil` value. Most other strategies use `PageMatch` internally, which only accepts an array of strings or a string from a `strategy` block and doesn't handle a `nil` value.

I'll create a `brew` PR in a bit to update the other strategies to handle a `nil` value, so we can simply use `next if ...` in `strategy blocks without needing to remember what's an appropriate default value for a given strategy.

**Comment by miccal** [comment]
Thank you @FnControlOption and @samford.

**Pull request merged**